ignite/contrib/metrics/regression/manhattan_distance.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import division
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+
+
+class ManhattanDistance(Metric):
+    r"""
+    Calculates the Manhattan Distance.
+
+    It has been proposed in `Performance Metrics (Error Measures) in Machine Learning Regression, Forecasting and
+    Prognostics: Properties and Typology`.
+
+    More details can be found in `https://arxiv.org/ftp/arxiv/papers/1809/1809.03006.pdf`.
+
+    :math:`\text{MD} = \sum _j^n (A_j - P_j)`
+
+    Where, :math:`A_j` is the ground truth and :math:`P_j` is the predicted value.
+
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y` and `y_pred` must be of same shape.
+
+    """
+    def reset(self):
+        self._sum_of_errors = 0.0
+        self._num_examples = 0
```

Comment:
@zasdfgbnm correct, unused variable, thanks for catching this !

Reply:
@vfdev-5 What is your opinion on the second part of my comment? i.e. about

> All metrics that has not been updated yet after last reset should raise `NotComputableError` when `compute`

Reply:
@vfdev-5 here for this case, instead of using `_num_examples`, it might be better to use a boolean variable called `_updated`, or even just remove the check (I don't know which choice is better).

Reply:
@zasdfgbnm this is a correct assumption. How would you implement this ?
A resetable counter of `update` method that is checked in `compute`... :|

Reply:
@vfdev-5 My suggestion is:
1. For the purpose of this PR, just remove `_num_examples` and don't worry about not being updated. If we are worried about this metric, then we should also be worried about more other metrics, so it worth a separate PR to correct all of them.
2. If we are using metrics only with engines, then it should be safe to assume a metric will never be computed before updated.
3. If we still want to check if it is updated before compute, some change in `Metric` like this might work:

```python
class Metric:
    def  __init__(self, ...):
        def wrapped_reset():
            self._updated = False
            self.reset()
        self.reset = functool.wraps(self.reset)(wrapped_reset)

        def wrapped_update(output):
            self._updated = True
            self.update(output)
        self.update =  functool.wraps(self.update)(wrapped_update)

        def wrapped_compute():
            if not self._updated:
                raise NotComputableError('not updated before compute')
            return self.compute()
        self.compute =  functool.wraps(self.compute)(wrapped_compute)
```

Reply:
For ManhattanDistance, since there is no division, it might be best to just return whatever the current value of self._sum_of_errors is. I have removed NotComputableError for this metric.